src/engine/Metavariable_pattern.ml

```diff
@@ -30,7 +30,9 @@ let logger = Logging.get_logger [ __MODULE__ ]
 
 let adjust_content_for_language (xlang : Xlang.t) (content : string) : string =
   match xlang with
-  | Xlang.L (Lang.Php, _) -> "<?php " ^ content
+  | Xlang.L (Lang.Php, _)
+    when not (content =~ {|[ \t\n]*<\?\(php\|=\)?[ \t\n]+|}) ->
+      "<?php " ^ content
```

Comment:
It's better than before but shouldn't this be built into the parser? Doesn't the pattern parser do something like this already?

I'm also wondering how come a closing `?>` isn't required. I guess it's valid PHP but it would be great to document it.

How about moving the logic to where PHP parsing happens. Something like this maybe:
```ocaml
let adjust_content_for_language (xlang : Xlang.t) (content : string) : string =
  match xlang with
  | Xlang.L (Lang.Php, prog_str) -> Lib_parsing_php.wrap_extracted_code prog_str
  | ...
```

Reply:
Good points, applied!